### PR TITLE
Blazor WebView alternate runtime platform

### PR DESCRIPTION
**Pull request by SteveSandersonMS**
Since @Eilon is [moving the WinForms/WPF support out into the MAUI repo](https://github.com/dotnet/aspnetcore/pull/33410), we're going to need some other way to run the Microsoft.AspNetCore.Blazor.WebView code that remains in this repo (and the corresponding `blazor.webview.js`) to make sure it actually works and we can add features to it.

This PR adds a new [Photino](https://www.tryphotino.io/) hosting platform which is a bit like the WinForms/WPF hosts except (1) it's cross-platform and can run on macOS/Linux too, and (2) it's limited to *purely* having a single Blazor webview in a window, and can't have any other native controls alongside it.

**There is no plan to ship this for public use**. People who actually want that can already use [Photino's built-in Blazor support](https://docs.tryphotino.io/Photino-Blazor) and simply don't need what we have here. The purpose of what we have here is just to help us validate and debug our ongoing work on Microsoft.AspNetCore.Blazor.WebView.

### CODE REVIEW

**Comment by Eilon** [comment]
Since the Photino stuff is just for "testing," should it go into some testing folder instead of Platform?

**Comment by Eilon** [comment]
(And then it doesn't need to track shipping APIs, etc.?)

**Comment by SteveSandersonMS** [comment]
I don’t know. It is a “platform” even though we’re not shipping it. It will be a sample/playground as much as it is for testing.

I marked the assemblies as nonshipping - is that sufficient?

**Comment by HaoK** [comment]
It is nice to not have sample/test code mixed in with production code to make it obvious that this isn't like all the other code, maybe its better in Components/WebView/samples directory that's already there?

**Review by TanayParikh — APPROVED** [review]

**Comment by Eilon** [comment]
@SteveSandersonMS I don't have a strong sense of what is best, but given that it doesn't ship at all, and is for testing/demo only, to me it seems to fit better in a samples/tests location.

**Comment by SteveSandersonMS** [comment]
OK, I've moved it under `Samples`. It's slightly quirky because there's a platform library, a test app, and (soon) a sample app. However I really don't feel strongly about it, so if this is what seems best to most people let's go that way.

**Comment by SteveSandersonMS** [comment]
@dotnet/aspnet-build The build is failing with:

```
Unable to find package Photino.NET. No packages exist with this id in source(s): dotnet-eng, dotnet-experimental, dotnet-public, dotnet-tools, dotnet31-transport, dotnet5, dotnet5-transport, dotnet6, dotnet6-transport, richnav
```

What's the latest story on adding a new 3rd-party NuGet package? I see we don't reference the public NuGet.org feed any more. I checked the `docs/` files and couldn't find anything describing the process for this. Is there some step needed to update one of our feeds?

FYI this new package is purely a sample/test dependency and doesn't impact any released code.

**Comment by SteveSandersonMS** [comment]
cc @dougbu @wtgodbe about https://github.com/dotnet/aspnetcore/pull/33413#issuecomment-858542791. Sounds like it has to be added to our feeds. Is this something you're able to do? Thanks!

**Review by Eilon — APPROVED** [review]

**Comment by wtgodbe** [comment]
Pushed

**Comment by Eilon** [comment]
@SteveSandersonMS there might still be a package missing: `Photino.Native`

**Comment by Eilon** [comment]
This one is the dependency of Photino.NET: https://www.nuget.org/packages/Photino.Native/

**Comment by wtgodbe** [comment]
> there might still be a package missing: Photino.Native

pushed

File: src/Components/ComponentsNoDeps.slnf [hunk 1]
```diff
@@ -37,6 +37,16 @@
       "src\\Components\\WebAssembly\\testassets\\Wasm.Authentication.Server\\Wasm.Authentication.Server.csproj",
       "src\\Components\\WebAssembly\\testassets\\Wasm.Authentication.Shared\\Wasm.Authentication.Shared.csproj",
       "src\\Components\\WebAssembly\\testassets\\WasmLinkerTest\\WasmLinkerTest.csproj",
+      "src\\Components\\WebView\\Samples\\PhotinoPlatform\\src\\Microsoft.AspNetCore.Components.WebView.Photino.csproj",
+      "src\\Components\\WebView\\Samples\\PhotinoPlatform\\testassets\\PhotinoTestApp\\PhotinoTestApp.csproj",
+      "src\\Components\\WebView\\Platforms\\WebView2\\src\\Microsoft.AspNetCore.Components.WebView.WebView2.csproj",
```
Comment (Eilon) [inline src/Components/ComponentsNoDeps.slnf:42]: Need to rebase?

Comment (SteveSandersonMS) [reply src/Components/ComponentsNoDeps.slnf]: Well spotted, thanks.

**Pull request merged**